_lab/lab07.md

```diff
@@ -113,32 +113,43 @@ def charactersPerWord(filename):
 def wordFrequency(filename):
     '''
     (20 points)
-    Reads the file from filename in your function and returns a dictionary 
+    Reads the file from filename in your function and returns a dictionary
     with the frequency of each word as its value.
     - Words are separated by whitespace characters, but do not include
     the following punctuation characters (,.!?;). You can assume contractions
     count as one word (i.e. "don't", "you'll", etc. are one word).
     - The split and strip functions may be useful.
-    - You can assume contractions count as one word 
+    - You can assume contractions count as one word
     (i.e. "don't", "you'll", etc. are one word).
     - Your function should open the file for reading, and close
     the file before returning.
     '''
     return "stub"
-    
-    
+
+
 def mostCommonWords(filename, N):
     '''
     (20 points)
     Reads the file from filename in your function and returns a list of N most
     common words in the text file (i.e., N words with the highest frequency),
     sorted by the number of times they occured in the file (most common first).
     - Use wordFrequency() helper function to count the frequency of each word.
-    - Print "[Error] The "<filename>" contains <X> unique words (you asked for <N>)." 
-    and return None if N is larger than the number of words in the file (substitute 
+    - Print "[Error] The "<filename>" contains <X> unique words (you asked for <N>)."
+    and return None if N is larger than the number of words in the file (substitute
     "<filename>", <X>, <N> with the actual values).
     '''
     return "stub"
 
 ```
 
+# Extra practice
+
+1. Write a function `set_intersection(A, B)` that takes two lists of numbers A and B as inputs, and returns a list containing all numbers common to both input lists. If a number occurs n times in both lists, it should occur n times in the result. The order of the result should be the order of occurrence in the first input list. Examples: `set_intersection([1, 8, 2, 1], [4, 1, 8])` should return `[1, 8]`; `set_intersection([8, 1, 2, 1], [4, 1, 8, 1])` should return `[8, 1, 1]`.
```

Comment:
The first and second problems are good practice exercises. I would call it `get_intersection`